### ISSUE TITLE
connect: cluster bootstrapped with Vault CA cannot issue certs

### ISSUE THREAD

**Issue by tradel** [hashicorp/consul#6269]
#### Overview of the Issue

If you bootstrap a new Consul cluster with Vault as the Connect CA, Consul will fail to generate any certificates because it doesn't correctly initialize the intermediate CA.

If you bootstrap with Consul's internal CA and then switch to Vault after leader election, it works as expected.

#### Reproduction Steps

1. Start Consul with Vault as the CA.
```
./consul agent -dev -hcl 'connect {enabled=true, ca_provider="vault", ca_config{token="redacted",root_pki_path="/root",intermediate_pki_path="/inter",address="http://localhost:8200"}}'
```

2. Try to issue a leaf cert.
```
curl localhost:8500/v1/agent/connect/ca/leaf/foobar
```

3. Observe the Consul output.
```
error issuing cert: Error making API request.

URL: PUT http://localhost:8200/v1/inter/sign/leaf-cert
Code: 404. Errors:

* no handler for route 'inter/sign/leaf-cert'
```

### Consul info for both Client and Server

Confirmed with both 1.5.3 and 1.6.0-beta3.

**Comment by i0rek** [comment]
@tradel can this issues be closed now?

**Comment by seguidor777** [comment]
Issue solved, it was a missconfiguration on the leaf cert from Vault. I would have liked to have a guide on how to configure Vault certs to make them work with Consul

Regards
Jorge Luna

**Comment by llchan** [comment]
I get the same panic as @seguidor777 above at https://github.com/hashicorp/consul/blob/9be6dfc310ddd54627d698537a98827245185290/agent/consul/connect_ca_endpoint.go#L540
and this should be handled gracefully even if vault is misconfigured.

I'll try to fiddle with the policy/etc to get this to work, and I agree a more detailed laundry list of the required permissions for each of the root/intermediate mounts would be helpful.

**Comment by ericbrumfield** [comment]
I just ran into this 'error having Vault cross-sign cert: Error making API request.' as well with a 403 response, but for a different reason I believe.  In my scenario I was updating the vault token that consul connect ca was using, which expired, AND was also attempting to rotate to a new vault pki mount for both the root and intermediary.

For this scenario, I was able to resolve the error by first updating consul connect ca config with a new and valid vault token.  Then I was able to change the vault pki mounts in a second config update, so my issue that got the same error message was when I attempted to change both the token and pki mount in 1 config update.  Our problem was not vault policy/permission related.

Our vault cluster has been operational for about 2 months now and this was a new error for me during token/pki mount rotation in a consul connect ca w/ vault setup.
